### PR TITLE
Separate community and enterprise subcommands for install/register/deregister

### PR DESCRIPTION
**Pull request by ryantking**
This separates the community and enterprise logic for the install, register, and deregister commands into `community` and `enterprise` subcommand. Unisntalling does not require any behavior changes beyond the release name, which can be set via flag.

BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh-enterprise/issues/149

### CODE REVIEW

**Review by joekelley — COMMENTED** [review]
We should also update our docs / guides to reflect these changes.

**Comment by ryantking** [comment]
@joekelley I updated all the commands, let me know if there's any actual copy you want written

File: pkg/meshctl/commands/cluster/register/cluster_register.go [hunk 1]
```diff
@@ -4,48 +4,120 @@ import (
...
+
+	opts.addToFlags(cmd.Flags())
+	cmd.SilenceUsage = true
+	return cmd
+}
+
+type enterpriseOptions options
+
+func (o *enterpriseOptions) addToFlags(flags *pflag.FlagSet) {
+	utils.AddAgentFlags(&o.AgentChartPathOverride, &o.AgentChartValues, flags, "Enterprise Agent", "enterprsie-agent-")
```
Comment (harveyxia) [inline pkg/meshctl/commands/cluster/register/cluster_register.go:122]:
```suggestion
	utils.AddAgentFlags(&o.AgentChartPathOverride, &o.AgentChartValues, flags, "Enterprise Agent", "enterprise-agent-")
```

File: pkg/meshctl/commands/cluster/register/cluster_register.go [hunk 2]
```diff
@@ -4,48 +4,120 @@ import (
...
-	set.StringVar(&opts.CertAgent.ChartValues, "cert-agent-chart-values", "", "Path to a Helm values.yaml file for customizing the installation of the Certificate Agent. If unset, this command will install the Certificate Agent with default Helm values.")
-	set.BoolVar(&opts.EnterpriseAgent.Install, "install-enterprise-agent", true, "If true, install the enterprise-agent on the cluster being registered if Enterprise Networking is detected.")
-	set.StringVar(&opts.EnterpriseAgent.ChartPath, "enterprise-agent-chart-file", "", "Path to a local Helm chart for installing the Enterprise Agent. If unset, this command will install the Wasm Agent from the publicly released Helm chart.")
-	set.StringVar(&opts.EnterpriseAgent.ChartValues, "enterprise-agent-chart-values", "", "Path to a Helm values.yaml file for customizing the installation of the Enterprise Agent. If unset, this command will install the Wasm Agent with default Helm values.")
+type communityOptions options
+
+func (o *communityOptions) addToFlags(flags *pflag.FlagSet) {
+	utils.AddAgentFlags(&o.AgentChartPathOverride, &o.AgentChartValues, flags, "Certificate Agent", "cert-agent-")
+}
```
Comment (harveyxia) [inline pkg/meshctl/commands/cluster/register/cluster_register.go:86]: these aren't used

File: pkg/meshctl/utils/flag_utils.go [hunk 3]
```diff
@@ -9,11 +11,31 @@ type GlobalFlags struct {
...
 // Set kubeconfig path and context flags for the management cluster.
 func AddManagementKubeconfigFlags(kubeconfig, kubecontext *string, flags *pflag.FlagSet) {
-	flags.StringVar(kubeconfig, "kubeconfig", "", "path to the kubeconfig from which the management cluster will be accessed")
-	flags.StringVar(kubecontext, "kubecontext", "", "name of the kubeconfig context to use for the management cluster")
+	flags.StringVar(kubeconfig, "kubeconfig", "", "Path to the kubeconfig from which the management cluster will be accessed")
+	flags.StringVar(kubecontext, "kubecontext", "", "Name of the kubeconfig context to use for the management cluster")
+}
+
+// AddAgentFlags adds the agent flags configured for the specific agent
+func AddAgentFlags(chartPath, valuesPath *string, flags *pflag.FlagSet, agentName, flagPrefix string) {
```
Comment (harveyxia) [inline pkg/meshctl/utils/flag_utils.go:24]: nit: i'd prefer doing without this indirection (and just using `flags.StringVar` for each command), it makes the logic a harder to understand without much deduplication

File: pkg/meshctl/commands/cluster/register/cluster_register.go [hunk 4]
```diff
@@ -4,48 +4,154 @@ import (
...
+	flags.StringVar(&o.Registration.RemoteNamespace, "remote-namespace", defaults.DefaultPodNamespace, "namespace in the target cluster where a service account enabling remote access will be created.\nIf the namespace does not exist it will be created.")
+	flags.StringVar(&o.Registration.APIServerAddress, "api-server-address", "", "Swap out the address of the remote cluster's k8s API server for the value of this flag.\nSet this flag when the address of the cluster domain used by the Gloo Mesh is different than that specified in the local kubeconfig.")
+	flags.StringVar(&o.Registration.ClusterDomain, "cluster-domain", "", "The Cluster Domain used by the Kubernetes DNS Service in the registered cluster. Defaults to 'cluster.local'.\nRead more: https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/")
+	flags.StringVar(&o.AgentCrdsChartPath, "agent-crds-chart-file", "", "Path to a local Helm chart for installing CRDs needed by remote agents.\nIf unset, this command will install the agent CRDs from the publicly released Helm chart.")
```
Comment (harveyxia) [inline pkg/meshctl/commands/cluster/register/cluster_register.go:50]: this is only needed for community

File: pkg/meshctl/commands/cluster/register/cluster_register.go [hunk 5]
```diff
@@ -4,48 +4,154 @@ import (
...
+}
+
+func communityCommand(ctx context.Context, regOpts *options) *cobra.Command {
+	opts := (*communityOptions)(regOpts)
+	cmd := &cobra.Command{
+		Use:   "community [cluster name]",
+		Short: "Register using the community certificate agent",
```
Comment (harveyxia) [inline pkg/meshctl/commands/cluster/register/cluster_register.go:57]:
```suggestion
		Short: "Register a cluster for Gloo Mesh community edition",
```

File: pkg/meshctl/commands/cluster/register/cluster_register.go [hunk 6]
```diff
@@ -4,48 +4,154 @@ import (
...
+		"Path to a Helm values.yaml file for customizing the installation of the Certificate Agent.\n"+
+			"If unset, this command will install the Certificate Agent with default Helm values.",
+	)
+}
+
+func enterpriseCommand(ctx context.Context, regOpts *options) *cobra.Command {
+	opts := (*enterpriseOptions)(regOpts)
+	cmd := &cobra.Command{
+		Use:   "enterprise [cluster name]",
+		Short: "Register using the enterprise agent",
```
Comment (harveyxia) [inline pkg/meshctl/commands/cluster/register/cluster_register.go:102]:
```suggestion
		Short: "Register a cluster for Gloo Mesh enterprise edition",
```

File: pkg/meshctl/commands/cluster/register/cluster_register.go [hunk 7]
```diff
@@ -4,48 +4,154 @@ import (
...
+		Example: `  # Register the current context
+  meshctl cluster register enterprise mgmt-cluster
+
+  # Register a different context when the current one is the management cluster
+  meshctl cluster register --remote-context=my-context enterprise remote-cluster`,
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			opts.Registration.ClusterName = args[0]
```
Comment (harveyxia) [inline pkg/meshctl/commands/cluster/register/cluster_register.go:110]: The logic contained in the registrant is specific for community cluster registration—Enterprise follows an entirely different flow. To review, the enterprise command needs to do the following:

1. Install the enterprise-agent Helm chart while exposing the Helm values [shown here](https://github.com/solo-io/gloo-mesh-enterprise/blob/master/enterprise-networking/ci/setup-networking.sh#L100-L112)

2. Create the KubernetesCluster CRDs as you're doing below

File: pkg/meshctl/commands/cluster/register/cluster_register.go (hunk 4 above)
Comment (ryantking) [reply pkg/meshctl/commands/cluster/register/cluster_register.go:48]: Enterprise does not need to install the agent CRDs?

File: pkg/meshctl/commands/install/install.go [hunk 8]
```diff
@@ -2,59 +2,302 @@ package install
...
 	}
+
+	return nil
+}
+
+func enterpriseCommand(ctx context.Context, installOpts *options) *cobra.Command {
+	opts := enterpriseOptions{options: installOpts}
+	cmd := &cobra.Command{
+		Use:   "enterprise",
+		Short: "Install the enterprise editionn (requires a license)",
```
Comment (joekelley) [inline pkg/meshctl/commands/install/install.go:204]: -> Install Gloo Mesh Enterprise

File: pkg/meshctl/commands/cluster/register/cluster_register.go [hunk 9]
```diff
@@ -4,48 +4,145 @@ import (
...
+	return cmd
+}
+
+type enterpriseOptions struct {
+	*options
+	relayServerAddress string
+}
+
+func (o *enterpriseOptions) addToFlags(flags *pflag.FlagSet) {
+	flags.StringVar(&o.relayServerAddress, "relay-server-address", "", "The address that the enterprise agentw will communicate with the relay server via.")
```
Comment (joekelley) [inline pkg/meshctl/commands/cluster/register/cluster_register.go:138]:
```suggestion
	flags.StringVar(&o.relayServerAddress, "relay-server-address", "", "The address via which the enterprise agent will communicate with the relay server.")
```

File: pkg/meshctl/enterprise/enterprise_utils.go [hunk 10]
```diff
@@ -2,38 +2,75 @@ package enterprise
...
+		KubeContext: opts.RemoteContext,
+		ChartUri:    chartPath,
+		Namespace:   opts.RemoteNamespace,
+		ReleaseName: gloomesh.EnterpriseAgentReleaseName,
+		ValuesFile:  opts.AgentChartValuesPath,
+		Verbose:     opts.Verbose,
+		Values: map[string]string{
+			"relay.serverAddress": opts.RelayServerAddress,
+			"relay.authority":     "enterprise-networking.gloo-mesh",
+			"relay.insecure":      "true",
```
Comment (harveyxia) [inline pkg/meshctl/enterprise/enterprise_utils.go:36]: this should be exposed to the user as a CLI flag

**Review by joekelley — APPROVED** [review]

**Pull request merged**